### DISCUSSION TITLE
Proposal feature : Multi-repository discussions

### DISCUSSION THREAD

**Discussion by scorsi** [community/community#41] [Discussions] [answered]
Hello,

Internally we're working on a big mono repository so we will not have the problem.

What if there's multiple repositories (in case of a micro-services system) and a discussion must be made for two of them ? 
May be, a multiple-repository discussion could be a good idea ? Or maybe, just showing or adding a reference of a discussion from another repository ? :thinking:

**Comment by martinamorris** [comment]
Another way to handle this would be an organization-wide discussion -- perhaps with specific repos tagged.

**Reply by scorsi** [reply]
That would be a solution :)

**Reply by encelo** [reply]
I support the idea of having an organization-wide Discussions tab. :+1:

**Reply by phoerious** [reply]
I would like to see this as well, but there are two things to consider:

- individual repositories should have a link to the org-level discussion forum, so it is easy to find (also for developers who may spend the majority of their time on the main repository's issue tracker)
- once this is implemented, there should be a migration path from repo level to project level

**Reply by haykam821** [reply]
In my opinion, an organization-wide discussion board should simply inherit the discussions from the organization's repos to limit the duplication of discussions, somewhat like a super-category.

**Reply by karnaker** [reply]
I support the idea of organization level discussions. Also, we should be able to add discussions to project boards, as the discussion may be part of the work. See #3775

**Comment by becca** [comment, marked as answer]
👋 hello there and thank you for this feedback! this is most certainly an area of feedback we're hearing from those in the beta as not everyone has a mono-repo structure. 

right now the way some communities are working around this is by opening a repository specifically for "community conversations" within their organization and directing traffic there. 

@martinamorris I'm curious how the latter solution would work (or doesn't) for y'all.

**Reply by TinoDidriksen** [reply]
> right now the way some communities are working around this is by opening a repository specifically for "community conversations" within their organization and directing traffic there.

Directing how? If there was built-in support for the per-repo Discussion link to be redirected to another repo, then that could work. But for orgs like us who have 564 repos and add new one every so often, this isn't really fun to administrate.

We would much rather have Discussions on the org page, and have every repo either go to the org Discussions, or optionally have their own Discussions where it makes sense (though I can't think of when it would for us).

**Reply by erikr** [reply]
Currently, Projects support multiple repos. Discussions could work in a similar manner.

**Reply by qwc** [reply]
So seriously for the Moving Blocks Organization with the main repo of MovingBlocks/terasology it would definitely make sense to have a Org wide Discussions possibility to cover all the topics and ideas not belonging to a specific repository and to be able to coordinate all the text resources covered throughout the org.

An idea here could also be to be able to put Discussions of org repos as sub-discussions into the org Discussions menu so to be able to jump there and back. Even links to non-org Discussion pages would make sense here, because terasology modules sometimes also live outside of the org.

**Reply by choldgraf** [reply]
We could definitely use this, as our organization has many modular projects that work together, but have different repositories. A few things that we'd really benefit from:

- The `Discussions` tab on a repository would direct to the org-level discussions board
- The `Convert to Discussion` button on an issue would be able to automatically convert it on the org-level discussions board

**Reply by endel** [reply]
If I could select multiple repositories on this page it would be perfect

https://github.com/organizations/[xxxxxxx]/settings/discussions

<img width="484" alt="image" src="https://user-images.githubusercontent.com/130494/236560419-f569d24d-3a86-4280-8768-a82f12a4a438.png">

**Reply by siminapasat** [reply]
@endel what's your use case to have multiple repositories, but not at organization level? Thanks for your insights! 😊

**Reply by endel** [reply]
Hi @siminapasat, thank you for your interest. 
I did enable it at [organization level](https://github.com/orgs/colyseus/discussions), and the "Discussions" link at the  [main repository](https://github.com/colyseus/colyseus) uses it. It would be helpful for my community to have the "Discussions" tab on other repos ([1](https://github.com/colyseus/colyseus.js), [2](https://github.com/colyseus/colyseus-unity-sdk), [3](https://github.com/colyseus/colyseus-haxe), [4](https://github.com/colyseus/colyseus-defold)) that also go to that same organization-level "discussions". (some users may not be aware only the "main" repo has the "discussions" link)

**Comment by martinamorris** [comment]
Thanks @becca!  I suppose we could do that.  Would it be possible to join your beta testing group so we could give this a run and see if there are any gotcha's?

**Reply by becca** [reply]
Is there a specific public repository you are targeting to have Discussions enabled on? 👀

**Reply by martinamorris** [reply]
Yes:  https://github.com/statnet/statnet.help

**Reply by becca** [reply]
Yes! I just flagged this repository in. You can now enable Discussions under the `/settings` tab

**Reply by martinamorris** [reply]
Thank you!

... and i already posted a question on feedback ;)

**Comment by gmondello** [comment]
chiming in here to add another customer perspective to this feature request. This would be enormously helpful for our company as it would allow us to sunset some other siloed Q&A platforms that developers don't really actively use any longer... 

Is this coming, or should we create a repo specifically for Q&A? This could probably work but feels awfully clunky.

**Reply by gmondello** [reply]
meant to tag @becca .. thx!

**Reply by gmondello** [reply]
i see she no longer works at GitHub though... hoping someone else from GitHub Product is watching!

**Reply by siminapasat** [reply]
Hey @gmondello ! Apologies for the _very_ late response. We launched this a while back https://github.blog/changelog/2022-04-12-organization-discussions/

**Comment by chiumichael** [comment]
Is work still being done on this feature?

**Reply by LindseyB** [reply]
Right now we've released [Organization Discussions](https://github.blog/changelog/2022-04-12-organization-discussions/) as a way to achieve this, but I understand that this might not fit your goals.
